### DISCUSSION TITLE
HTTP error handling   in Gatsby site

### DISCUSSION THREAD

**Discussion by sheetal-addweb** [gatsbyjs/gatsby#29008] [Help]
How to handle HTTP error and redirect on custom page base on error status code for example 403,404,502 etc...

**Comment by ascorbic** [comment]
Hi. You can [add a 404 page](https://www.gatsbyjs.com/docs/how-to/adding-common-features/add-404-page/) like this, but the others would require handling at the server or proxy level. Depending on your host or CDN you might be able to configure them, but that will depend on each host.

**Reply by kay-o** [reply]
Hi @sheetal-addweb : what are the conditions in which you are seeing the HTTP error? Are you seeing a 404 error page when you would expect to see one, e.g. a 404 when you enter a correctly formed url that simply doesn't have a corresponding page? Or are you seeing unexpected results when you should see a 404?

Since your question sounds like one you might have answered with existing documentation, you may be facing a unique situation. Interested to know more.

**Reply by sheetal-addweb** [reply]
Hi @kay-o  In my case if user added correct formatted URL but corresponding page is not available in that case need to show 404 page
